contracts: tests for OsmoticController

0xGabi opened the pull request:
closes #17

0xGabi committed:
contracts: include initial tests for OsmoticController

0xGabi committed:
contracts: generalize initial setup deployment to their own BaseSetup contract

0xGabi committed:
chore: update submodule to our own fork

0xGabi committed:
contracts: include OsmoticController dependencies

PJColombo reviewed (commented):
Let's move the `/test/BaseSetup` file to the `/script` folder

kanaagawa commented on pkg/contracts/script/BaseSetup.sol:
```diff
@@ -46,6 +46,7 @@ contract BaseSetup is SetupScript {
     function setUp() public virtual {
         // create forks
         goerliFork = vm.createFork(GOERLI_RPC_URL);
+        vm.selectFork(goerliFork);
```
Is important to choose the fork right away so our contracts exists on the forked chain.

0xGabi committed:
contracts: move BaseSetup to scripts folder + finish lockBalance test

0xGabi committed:
contracts: this pr does:
- Update default verbosity level of tests to 1
- Modify BaseSetup to configure controller with the new constructor params
- Update Controller and Pool
- Fix broken tests

0xGabi committed:
contracts: OsmoticController add remaining tests

0xGabi committed:
contracts: integrate MimeToken logic to implementation (#20)

* contracts: update base setup scripts and tests

* contracts: integrate MimeTokenFactory logic + add createPool function

* contracts: update mime-token implementation

* contracts: update tests after mime-token changes

* contracts: include review changes and new mime-token updates

0xGabi committed:
Merge branch 'tests' into controller-tests
